laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -317,75 +308,31 @@ public function parseCaseProcedure($case)
-        $resident_data = Resident::where("email", $_SERVER["HTTP_EMAIL"])->get();
-        $resident = $resident_data[0]["id"];
+        $current_resident = Resident::where("email", $_SERVER["HTTP_EMAIL"])->get();
+        $resident = $current_resident[0]["id"];
 
         // id is stored as id1_id2_id3, need to split it to get the individual ids
         $split = explode("_", $id);
 
-        //Initiate first choice data
-        $data1 = [
-            "schedule" => null,
-            "attending" => null,
-        ];
-        // get information for first choice
-        $choice = 1;
-        $schedule_data1 = ScheduleData::where("id", $split[0])->get();
-        $attending_string = $schedule_data1[0]["lead_surgeon"];
-        $attending = substr(
-            $attending_string,
-            strpos($attending_string, "[") + 1,
-            strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
-        );
-        $attending1 = substr($attending_string, 0, strpos($attending_string, "["));
-        // Store first choice data
-        $date = $schedule_data1[0]["date"];
-        $data1["schedule"] = $schedule_data1[0];
-        $data1["attending"] = $attending1;
-
-        //Initiate second choice data
-        $data2 = [
-            "schedule" => null,
-            "attending" => null,
-        ];
-        $choice++;
-        // If the second choice exists, get information for second choice
-        if ($split[1] != 0) {
-            $schedule_data2 = ScheduleData::where("id", $split[1])->get();
-            $attending_string = $schedule_data2[0]["lead_surgeon"];
-            $attending = substr(
-                $attending_string,
-                strpos($attending_string, "[") + 1,
-                strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
-            );
-            $attending2 = substr($attending_string, 0, strpos($attending_string, "["));
-            // Store second choice data
-            $data2["schedule"] = $schedule_data2[0];
-            $data2["attending"] = $attending2;
-        }
-
-        //Initiate second choice data
-        $data3 = [
-            "schedule" => null,
-            "attending" => null,
-        ];
-        $choice++;
-        // If the third choice exists, get information for third choice
-        if ($split[2] != 0) {
-            $schedule_data3 = ScheduleData::where("id", $split[2])->get();
-            $attending_string = $schedule_data3[0]["lead_surgeon"];
-            $attending = substr(
-                $attending_string,
-                strpos($attending_string, "[") + 1,
-                strpos($attending_string, "]") - (strpos($attending_string, "[") + 1)
-            );
-            $attending3 = substr($attending_string, 0, strpos($attending_string, "["));
-            // Store second choice data
-            $data3["schedule"] = $schedule_data3[0];
-            $data3["attending"] = $attending3;
+        for ($i = 0; $i < 3; $i++) {
```

Comment:
I'll change the variable name and I'll push the resident_data array instead of reassigning it, but a foreach loop would introduce a little more logic because I think there needs to be some null elements of the residend_data to avoid index errors in the views. I think with how soon we are merging cleanup with prod we should stick with the for loops.

Reply:
I'd rather delay deploying the changes of this PR than merge it before it's finished.

Reply:
I still think that the for loop should stay. A foreach would be better if the number of preferences changed a lot, but barring any foundational change in the way that REMODEL works (ie asking the resident for more preferences), then this is still the best solution imo.

Reply:
While I agree that it's unlikely, the restriction of 3 preferences could change in the future.

If we can avoid using magic numbers, we should. They make code significantly less flexible to change.